metrix-integration/src/main/java/com/powsybl/metrix/integration/AbstractMetrix.java

```diff
@@ -85,32 +67,17 @@ protected AbstractMetrix(Reader remedialActionsReader, ReadOnlyTimeSeriesStore s
         this.logArchive = logArchive;
         this.computationManager = Objects.requireNonNull(computationManager);
         this.appLogger = Objects.requireNonNull(appLogger);
-        Objects.requireNonNull(analysisResult);
-        this.contingenciesProvider = network -> analysisResult.contingencies;
-        this.mappingConfig = analysisResult.mappingConfig;
-        this.metrixDslData = analysisResult.metrixDslData;
-        this.network = analysisResult.network;
-        this.metrixParameters = analysisResult.metrixParameters;
-        this.mappingParameters = analysisResult.mappingParameters;
-    }
-
-    protected static void compress(Reader reader, WorkingDirectory directory, String fileNameGz) {
-        try (BufferedReader bufferedReader = new BufferedReader(reader);
-             Writer writer = new OutputStreamWriter(new GZIPOutputStream(new BufferedOutputStream(Files.newOutputStream(directory.toPath().resolve(fileNameGz)))), StandardCharsets.UTF_8)) {
-            CharStreams.copy(bufferedReader, writer);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
+        this.analysisResult = Objects.requireNonNull(analysisResult);
     }
 
-    protected static int computeChunkSize(MetrixRunParameters runParameters, MetrixConfig metrixConfig, TimeSeriesIndex index) {
+    protected static int computeChunkSize(MetrixRunParameters runParameters, int chunkSizeFromConfig, TimeSeriesIndex index) {
         if (runParameters.getChunkSize() != -1 && runParameters.getChunkSize() < index.getPointCount()) {
             return runParameters.getChunkSize();
-        } else if (metrixConfig.getChunkSize() != -1 && metrixConfig.getChunkSize() < index.getPointCount()) {
-            return metrixConfig.getChunkSize();
-        } else {
-            return index.getPointCount();
         }
+        if (chunkSizeFromConfig != -1 && chunkSizeFromConfig < index.getPointCount()) {
+            return chunkSizeFromConfig;
+        }
+        return index.getPointCount();
```

Comment:
You can probably replace this with something like:
```suggestion
        return chunkSizeFromConfig != -1 && chunkSizeFromConfig < index.getPointCount() ? chunkSizeFromConfig : index.getPointCount();
```